core/settlement/engine.go

```diff
@@ -309,18 +311,18 @@ func (e *Engine) SettleMTM(ctx context.Context, markPrice *num.Uint, positions [
 		// and the old mark price at which the party held the position
 		// the trades slice contains all trade positions (position changes for the party)
 		// at their exact trade price, so we can MTM that volume correctly, too
-		mtmShare, mtmDShare, neg := calcMTM(markPrice, current.price, current.size, traded, e.positionFactor)
+		mtmShare, mtmDShare, neg := calcMTM(markPrice, lastSettledPrice, current, traded, e.positionFactor)
```

Comment:
Not sure if we can always use `lastSettledPrice` here, unless we make this call for each trade, and set the `lastSettledPrice` to the trade price. I'd have to take a closer look, but just thought I'd mention it as I go through this

Reply:
I didn't change the behaviour - it's doing exactly what it did before. whether this was correct I didn't evaluate. I was trying to keep things working as they were.

Reply:
Hmm, we pass in the trades, those should be handled there. Ignore this comment, was a bit too trigger-happy